build_web_compilers/lib/src/dev_compiler_builder.dart

```diff
@@ -238,6 +246,15 @@ Future<void> _createDevCompilerModule(
       var json = jsonDecode(content);
       json['sources'] = fixSourceMapSources((json['sources'] as List).cast());
       await buildStep.writeAsString(sourceMapId, jsonEncode(json));
+
+      // remove temp directories from metadata
```

Comment:
nit: I would be a bit more clear here:

```dart
// Copy the metadata output, modifying its contents to remove the temp directory from paths
```

Also, I would expect to see the `multi-root` scheme or something here, the metadata files might need to be updated - this will be harder to clean up in blaze I think (there isn't just one possible dir to strip out).

Reply:
Is the same code running in blaze?

Reply:
No `build_web_compilers` is external only. The blaze integration uses custom starlark build rules, it doesn't go through the `Builder` interface.

Reply:
Right, I just mean that if you are having to write this here it will be harder to replicate there.

Reply:
I need to change the ddc code to generate metadata with a multiroot scheme, currently it has full paths inside, which is not what we what for part of build output. 

Created an issue to modify metadata: https://github.com/dart-lang/sdk/issues/43309
Created an issue to update the code according to the change in build_web_compilers: https://github.com/dart-lang/build/issues/2802